### ISSUE TITLE
Gaps in way spells are handled

### ISSUE THREAD

**Issue by cswendrowski** [foundryvtt/pf2e#547]
In GitLab by @Shandyan on Jun 18, 2020, 04:18

After checking through the spells for errors, I've identified a few gaps in the spells are currently handled. 

**Spell damage**

At present, each spell can only do one damage type and amount (e.g. 5d6 fire). However, there are a number of spells which do two damage types (e.g. searing light), a random damage type (e.g. abyssal wrath), damage that varies with choices the caster makes (e.g. divine lance, ki strike), or damage that only affects some targets (e.g. chilling darkness, harm).  If we could use the same damage system as for attacks, this would help with a lot of cases. 

For extra complexity, some spells do damage of type X which counts as type Y for resistances (e.g. moonbeam or ephemeral hazards), and some do bonus damage based on a non-casting attribute (e.g. hurtling stone). 

**Critical Effects**

Unlike strikes, spells don't always just do double damage on a critical. They can do entirely new conditions (e.g. produce flame) or increase damage values without strictly doubling.  One way around this would be to separately enter the critical failure, failure, success and critical success entries for each spell.

**Spell Attacks & Saves**

A few spells require both a successful spell attack and for the target to make a save (e.g. ray of enfeeblement and disintegrate). At the moment, a spell can be either an attack or a save, not both.

**Spell Areas**

There are several spells that can be in one or two different areas (e.g. dragon breath, illusory terrain). Some spells gain increased areas as they are heightened. Would it possible to let spells have multiple entries for area? 

**Variable Action Spells**

These aren't handled at all by the system right now. Increasing the actions spent can simply increase damage (e.g. magic missile) or the area (e.g. illusory terrain), but harm and heal are probably the two most complicated.

**Comment by TMunsie** [comment]
@CarlosFdez Can this be closed?

**Comment by SpartanCPA** [comment]
@Shandyan how much of this is still not implemented?

**Comment by CarlosFdez** [comment]
Just noticed this sorry. This is a full feature list that leads to the entire thing not really be closeable, and it should be split apart.

Spell areas might be further out and might need its own issue. Variable Action Spells is almost available (waiting on stwlam for a second opinion). Spell Attacks & Saves might need their own issue but wouldn't be that hard to do. Critical Effects might be hard to do.

Spell Damage is done outside of caster choice damage types.
